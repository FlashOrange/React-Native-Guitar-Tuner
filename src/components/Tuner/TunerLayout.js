import React, { Component } from 'react';
import Aux from '../../hoc/Aux';
import SettingsButton from './TunerComponents/SettingsButton';
import TunerDisplay from './TunerComponents/TunerDisplay';
import Strings from './TunerComponents/Strings';

const TunerLayout = props => {
  return (
    <Aux>
      <SettingsButton
        text={props.instrumentName}
        pressed={props.onSettingsButtonPress}
      />
      <TunerDisplay delta={props.delta} />
      <Strings
        stringNotes={props.stringNotes}
        activeString={props.activeString}
      />
    </Aux>
  );
};

export default TunerLayout;
