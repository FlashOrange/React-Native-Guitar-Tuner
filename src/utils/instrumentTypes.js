const guitar = {
  displayName: 'Guitar',
  stringNotes: ['E', 'A', 'D', 'G', 'B', 'E'],
  stringFrequencies: [
    { min: 36, max: 96, tuned: 82 },
    { min: 96, max: 128, tuned: 110 },
    { min: 128, max: 171, tuned: 146 },
    { min: 171, max: 221, tuned: 196 },
    { min: 221, max: 287, tuned: 246 },
    { min: 287, max: 371, tuned: 329 },
  ],
};

const bass4 = {
  displayName: 'Bass (4-String)',
  stringNotes: ['E', 'A', 'D', 'G'],
  stringFrequencies: [
    { min: 30, max: 48.55, tuned: 41.2 },
    { min: 48.56, max: 64.21, tuned: 55 },
    { min: 64.22, max: 86.61, tuned: 73.42 },
    { min: 86.62, max: 110, tuned: 98 },
  ],
};

const bass5low = {
  displayName: 'Bass (5-String, Low-B)',
  stringNotes: ['B', 'E', 'A', 'D', 'G'],
  stringFrequencies: [
    { min: 20, max: 36.48, tuned: 30.87 },
    { min: 36.49, max: 48.55, tuned: 42.1 },
    { min: 48.56, max: 64.21, tuned: 55 },
    { min: 64.22, max: 86.61, tuned: 73.42 },
    { min: 86.62, max: 140, tuned: 98 },
  ],
};

const bass5hi = {
  displayName: 'Bass (5-String, High-C)',
  stringNotes: ['E', 'A', 'D', 'G', 'C'],
  stringFrequencies: [
    { min: 36.49, max: 48.55, tuned: 41.2 },
    { min: 48.56, max: 64.21, tuned: 55 },
    { min: 64.22, max: 86.61, tuned: 73.42 },
    { min: 86.62, max: 114.4, tuned: 98 },
    { min: 114.41, max: 150, tuned: 130.81 },
  ],
};

const ukulele = {
  displayName: 'Ukulele',
  stringNotes: ['G', 'C', 'E', 'A'],
  stringFrequencies: [
    { min: 360.85, max: 416, tuned: 392 },
    { min: 326.81, max: 295.36, tuned: 261.63 },
    { min: 295.37, max: 360.84, tuned: 329.63 },
    { min: 416.1, max: 114.4, tuned: 440 },
  ],
};

const instrumentTypes = {
  guitar,
  bass4,
  bass5low,
  bass5hi,
  ukulele,
};

export default instrumentTypes;
