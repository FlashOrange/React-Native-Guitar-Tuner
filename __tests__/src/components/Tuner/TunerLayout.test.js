import React from 'react';
import 'react-native';
import TunerLayout from '../../../../src/components/Tuner/TunerLayout';
import renderer from 'react-test-renderer';
import instrumentTypes from '../../../../src/utils/instrumentTypes';

const instrument = instrumentTypes[Object.keys(instrumentTypes)[0]];

const config = {
  stringNotes: instrument.stringNotes,
  instrumentName: instrument.displayName,
};

test('it renders correctly', () => {
  const tree = renderer.create(<TunerLayout {...config} />).toJSON();
  expect(tree).toMatchSnapshot();
});
