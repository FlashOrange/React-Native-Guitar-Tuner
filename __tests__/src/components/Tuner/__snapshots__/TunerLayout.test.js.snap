// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders correctly 1`] = `
Array [
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "center",
        "marginBottom": 4,
        "opacity": 1,
        "padding": 8,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "white",
          "fontSize": 25,
          "fontWeight": "bold",
          "marginRight": 10,
          "textAlign": "center",
        }
      }
    >
      Guitar
    </Text>
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "white",
            "fontSize": 30,
          },
          Object {
            "alignSelf": "center",
            "marginLeft": 10,
            "marginTop": 2,
          },
          Object {
            "fontFamily": "FontAwesome",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      ïƒ—
    </Text>
  </View>,
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#1f2025",
        "marginBottom": 28,
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../img/tuner.jpg",
        }
      }
      style={
        Object {
          "resizeMode": "contain",
          "width": 750,
        }
      }
    />
    <Image
      collapsable={undefined}
      source={
        Object {
          "testUri": "../../../img/indicator.jpg",
        }
      }
      style={
        Object {
          "left": 375,
          "position": "absolute",
          "top": 10,
        }
      }
    />
  </View>,
  <View
    style={
      Object {
        "borderTopColor": "#3db78b",
        "borderTopWidth": 5,
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "space-around",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../img/string.jpg",
          }
        }
        style={
          Object {
            "flex": 1,
          }
        }
      />
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#1f2025",
              "borderColor": "#f3c556",
              "borderRadius": 25,
              "borderWidth": 3,
              "height": 50,
              "padding": 10,
              "position": "absolute",
              "top": 50,
              "width": 50,
            },
            Object {
              "borderColor": "#3db78b",
            },
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "white",
              "fontSize": 19,
              "textAlign": "center",
            }
          }
        >
          E
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../img/string.jpg",
          }
        }
        style={
          Object {
            "flex": 1,
          }
        }
      />
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#1f2025",
              "borderColor": "#f3c556",
              "borderRadius": 25,
              "borderWidth": 3,
              "height": 50,
              "padding": 10,
              "position": "absolute",
              "top": 50,
              "width": 50,
            },
            Object {
              "borderColor": "#3db78b",
            },
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "white",
              "fontSize": 19,
              "textAlign": "center",
            }
          }
        >
          A
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../img/string.jpg",
          }
        }
        style={
          Object {
            "flex": 1,
          }
        }
      />
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#1f2025",
              "borderColor": "#f3c556",
              "borderRadius": 25,
              "borderWidth": 3,
              "height": 50,
              "padding": 10,
              "position": "absolute",
              "top": 50,
              "width": 50,
            },
            Object {
              "borderColor": "#3db78b",
            },
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "white",
              "fontSize": 19,
              "textAlign": "center",
            }
          }
        >
          D
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../img/string.jpg",
          }
        }
        style={
          Object {
            "flex": 1,
          }
        }
      />
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#1f2025",
              "borderColor": "#f3c556",
              "borderRadius": 25,
              "borderWidth": 3,
              "height": 50,
              "padding": 10,
              "position": "absolute",
              "top": 50,
              "width": 50,
            },
            Object {
              "borderColor": "#3db78b",
            },
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "white",
              "fontSize": 19,
              "textAlign": "center",
            }
          }
        >
          G
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../img/string.jpg",
          }
        }
        style={
          Object {
            "flex": 1,
          }
        }
      />
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#1f2025",
              "borderColor": "#f3c556",
              "borderRadius": 25,
              "borderWidth": 3,
              "height": 50,
              "padding": 10,
              "position": "absolute",
              "top": 50,
              "width": 50,
            },
            Object {
              "borderColor": "#3db78b",
            },
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "white",
              "fontSize": 19,
              "textAlign": "center",
            }
          }
        >
          B
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../img/string.jpg",
          }
        }
        style={
          Object {
            "flex": 1,
          }
        }
      />
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#1f2025",
              "borderColor": "#f3c556",
              "borderRadius": 25,
              "borderWidth": 3,
              "height": 50,
              "padding": 10,
              "position": "absolute",
              "top": 50,
              "width": 50,
            },
            Object {
              "borderColor": "#3db78b",
            },
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "white",
              "fontSize": 19,
              "textAlign": "center",
            }
          }
        >
          E
        </Text>
      </View>
    </View>
  </View>,
]
`;
